tools/cordova/package-id-version-parser.js

```diff
@@ -0,0 +1,27 @@
+// Accepts a combined Cordova package ID + version string, then parses out
+// and returns the ID and version in a package details object.
+//
+// Example `packageIdAndVersion` formats:
+// some-cordova-plugin@1.0.0
+// @somescope/some-cordova-plugin@1.0.0
+exports.parse = packageIdAndVersion => {
+  const package = {};
+  if (packageIdAndVersion) {
+    const parts = packageIdAndVersion.match(
+      /^(@[^\/]*)?(\/)?([^@]*)(@)?(.*)?/
```

Comment:
```js
/^(@[^\/]*)?(\/)?([^@]*)(@)?(.*)?/
            ^^^^^
```
Could this be included in the first grouping along with the scope and not explicitly captured? Right now this allows `/dd@` and puts `/` into `scopeSeparator` with no `scope`.

Maybe...
```js
/^(@[^\/]*\/?)?([^\/@]*)(@)?(.*)?/
```
... you'll note I also added `/` to the negated character class.

Reply:
I split the separator out intentionally; I just liked the idea of having buckets for each individual component, so the scope (when available) was the actual scope name without the trailing `/`. But, there really isn't any reason to do this, so your point is well taken - changes coming!